src/ModelContextProtocol/Protocol/Transport/StdioServerTransport.cs

```diff
@@ -73,4 +73,32 @@ private static string GetServerName(McpServerOptions serverOptions)
 
         return serverOptions.ServerInfo?.Name ?? McpServer.DefaultImplementation.Name;
     }
+
+    // Neither WindowsConsoleStream nor UnixConsoleStream respect CancellationTokens or cancel any I/O on Dispose.
```

Comment:
I take it these are the platform-specific implementations from System.Console? Is there a bug we need to be reporting?

Reply:
I wasn't sure. Canceling a stdin read and then resuming it without exiting the process seemed like a niche scenario, so I wasn't going to raise a stink if we felt this workaround was good enough for tearing down the host and process, but it looks like @stephentoub is trying to make some improvements with https://github.com/dotnet/runtime/pull/114378.